File: app/src/main/java/com/wildlifeoftianjin/ui/framework/recyclerview/RecordListAdapter.java
package com.wildlifeoftianjin.ui.framework.recyclerview;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.wildlifeoftianjin.R;
import com.wildlifeoftianjin.model.RecordOverview;

/**
 * Created by presisco on 2017/5/17.
 */

public class RecordListAdapter extends DragAppendAdapter<RecordOverview> implements DragAppendAdapter.ActionListener {
    private RecordListener mListener;

    public RecordListAdapter(Context context, FooterListener listener, RecordListener recordListener) {
        super(context);
        setActionListener(this);
        setFooterListener(listener);
        mListener = recordListener;
    }

    @Override
    public ContentHolder onCreate(LayoutInflater inflater, ViewGroup parent) {
        return new RecordHolder(inflater.inflate(R.layout.listitem_record_overview, parent, false), mListener);
    }

    @Override
    public void onBind(ContentHolder holder, int position) {
        RecordHolder recordHolder = (RecordHolder) holder;
        RecordOverview overview = getItem(position);
        recordHolder.timeText.setText(overview.getTime());
        recordHolder.locationText.setText(overview.getLocation());
    }

    @Override
    public void onRecycled(ContentHolder holder) {

    }

    public interface RecordListener {
        void onClickRecord(int position);
    }

    protected static class RecordHolder extends ContentHolder {
        public final TextView timeText;
        public final TextView locationText;
        public final TextView countText;

        public RecordHolder(View itemView, final RecordListener listener) {
            super(itemView);
            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    listener.onClickRecord(getAdapterPosition());
                }
            });
            timeText = (TextView) itemView.findViewById(R.id.textTime);
            locationText = (TextView) itemView.findViewById(R.id.textLocation);
            countText = (TextView) itemView.findViewById(R.id.textCount);
        }
    }
}

File: app/src/main/java/com/wildlifeoftianjin/ui/framework/recyclerview/RecordListAdapter.kt
package com.wildlifeoftianjin.ui.framework.recyclerview

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.wildlifeoftianjin.R
import com.wildlifeoftianjin.model.RecordOverview

/**
 * Created by presisco on 2017/5/17.
 */

class RecordListAdapter(context: Context, onFooter: () -> Unit, private val onClickedRecord: (position: Int) -> Unit)
    : DragAppendAdapter.ActionListener, DragAppendAdapter<RecordOverview>(context) {

    init {
        onAction = this
        super.onFooter = onFooter
    }

    override fun onCreate(inflater: LayoutInflater, parent: ViewGroup): DragAppendAdapter.ContentHolder {
        return RecordHolder(inflater.inflate(R.layout.listitem_record_overview, parent, false), onClickedRecord)
    }

    override fun onBind(holder: DragAppendAdapter.ContentHolder, position: Int) {
        val recordHolder = holder as RecordHolder
        val overview = getItem(position)
        recordHolder.timeText.text = overview.getTime()
        recordHolder.locationText.text = overview.getLocation()
        recordHolder.countText.text = overview.getCount()
        recordHolder.headerText.text = overview.getUsername()
    }

    override fun onRecycled(holder: DragAppendAdapter.ContentHolder) {

    }

    class RecordHolder(itemView: View, listener: (position: Int) -> Unit) : DragAppendAdapter.ContentHolder(itemView) {
        val timeText: TextView = itemView.findViewById(R.id.textTime) as TextView
        val locationText: TextView = itemView.findViewById(R.id.textLocation) as TextView
        val countText: TextView = itemView.findViewById(R.id.textCount) as TextView
        val headerText: TextView = itemView.findViewById(R.id.textHeader) as TextView

        init {
            itemView.setOnClickListener { listener(adapterPosition) }
        }
    }
}
